src/test/java/org/example/Operations/DataObjectOperationsTest.java

```diff
@@ -512,188 +545,172 @@ public void test_modifying_metadata_atomically() {
+            firstRowSecondColumn = rootNode.path("rows").get(0).get(1).asInt();
 
             assertEquals(0, firstRowFirstColumn);
             assertEquals(15, firstRowSecondColumn);
 
-        } catch (Exception e) {
-            e.printStackTrace();
+            // Remove the data object.
+            DataObjectRemoveParams params = new DataObjectRemoveParams();
+            params.setNoTrash(1);
+            params.setAdmin(1);
+            res = rods.dataObject().remove(rodsToken, dataObject, 0, params);
+            assertEquals("remove data object request failed", 200, res.getHttpStatusCode());
+            assertEquals("remove data object failed", 0,
+                    getIrodsResponseStatusCode(res.getBody()));
+        } finally {
+            // Remove the data object.
+            DataObjectRemoveParams params = new DataObjectRemoveParams();
+            params.setNoTrash(1);
+            params.setAdmin(1);
+            rods.dataObject().remove(rodsToken, dataObject, 0, params);
         }
-
-        // Remove the data object.
-        DataObjectRemoveParams params = new DataObjectRemoveParams();
-        params.setNoTrash(1);
-        params.setAdmin(1);
-        res = rods.dataObject().remove(token, data_object, 0, params);
-        assertEquals("remove data object request failed", 200, res.getHttpStatusCode());
-        assertEquals("remove data object failed", 0,
-                getIrodsResponseStatusCode(res.getBody()));
     }
 
     private int getIrodsResponseStatusCode(String jsonString) {
-        try {
-            JsonNode rootNode = mapper.readTree(jsonString);
-            JsonNode statusCodeNode = rootNode.path("irods_response").path("status_code");
-            return statusCodeNode.asInt();
-        } catch (Exception e) {
-            e.printStackTrace();
-        }
-        return -1;
-    }
-
-    private void resetPath(Wrapper wrapper, String token, String lpath) {
-        res = wrapper.collections().list(token, lpath, null);
-
-        List<String> entries = new ArrayList<>();
-        try {
-            JsonNode rootNode = mapper.readTree(res.getBody());
-            JsonNode entriesNode = rootNode.path("entries");
-
-            for (JsonNode node : entriesNode) {
-                entries.add(node.asText());
-            }
-
-        } catch (IOException e) {
-            e.printStackTrace();
-        }
-
-        if (entries != null && !entries.isEmpty()) {
-            for (String entry : entries) {
-                wrapper.collections().remove(token, entry, null);
-                wrapper.dataObject().remove(token, entry, 0, null);
-                wrapper.resourceOperations().remove(token, entry);
-            }
-        }
+        JsonNode rootNode = assertDoesNotThrow(() -> mapper.readTree(jsonString),
+                "JsonProcessingException was thrown");
+        JsonNode statusCodeNode = rootNode.path("irods_response").path("status_code");
+        return statusCodeNode.asInt();
```

Comment:
I'm pretty sure I saw this in the other PR.

Open an issue to move this into a utility/support java file so it can be shared/used in multiple tests.

You can address the issue in a separate PR.

Reply:
Created #54